DOCSP-33948: Atomic typos and cleanup (#829)
mongodb/docs-node: source/includes/localhost-connection.rst

> If you want to run a MongoDB server on your local machine instead of
> using an Atlas cluster, complete the following steps:

Reviewer: S: the intro text for this section (on the actual docs page) makes it clear that this is an alternative to Atlas

there was also a ticket for Java(?) some time ago that encouraged "MongoDB Server" as a product name and "MongoDB deployment" for individual instances, iirc

```suggestion
To connect to a MongoDB deployment on your local machine,
complete the following steps:
```
